### PR TITLE
Pause Container Packaging

### PR DESCRIPTION
**Pull request by vsiddharth**
### Summary

Packaging Pause-Container

### Implementation details

The pause-container tarball is exported to the `out` directory.
This tarball is further stuck into the `/images` within the agent tarball. 

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 
No

Changes were manually tested. 

### Description for the changelog

* Packaging Pause Container for amazon-ecs-agent
### Licensing

This contribution is under the terms of the Apache 2.0 License:  
Yes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,7 +11,11 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
-.PHONY: all gobuild static docker release certs test clean netkitten test-registry run-functional-tests gremlin benchmark-test gogenerate run-integ-tests image-cleanup-test-images
+PAUSE_CONTAINER_IMAGE = "amazon/amazon-ecs-pause"
+PAUSE_CONTAINER_TAG = "0.1.0"
```
Comment (aaithal) [inline Makefile]: How are you planning to surface this tag information to the Agent when it loads this image?

Comment (vsiddharth) [inline Makefile]: The LoadImage API uses [LoadImageOptions](https://github.com/fsouza/go-dockerclient/blob/6936859b3576e8f5abd069c0edbcd454133632a2/image.go#L342) and there isn't an explicit way to expose the tag information there. 

The tag information is persisted in the pause-container tarball which is restored upon import.

Comment (aaithal) [inline Makefile]: I was asking about the logic of the code where we need to create a container using this image, apologies for the lack of clarity earlier.

To rephrase, how are you planning to surface this tag information to the Agent when it has to create a container using this image? Should we always use a constant tag (Example: `latest`)? Or should that be in code somewhere (which means we update with every release of pause container)? What are the pros/cons for these?

Comment (vsiddharth) [inline Makefile]: As the path to the pause-container tarball is fixed, I was thinking along the lines of having a constant in our config. We could also add additional fields to our config struct to hold these  attributes:
* PauseContainerTarballPath
* PauseContainerTag 

However, I was thinking of following up with another PR for loading pause container on startup. This PR is just for packaging the pause-container tarball within the agent image.

Comment (aaithal) [inline Makefile]: alright, sgtm

**Pull request merged**